### ISSUE TITLE
preProcessFile does not detect "import type" statements

### ISSUE THREAD

**Issue by kitsonk** [microsoft/TypeScript#37338]
**TypeScript Version:**  3.8.3

**Search Terms:**

preProcessFile, type only import, ambientExternalModules

**Code**

```ts
import ts from "typescript";

const text = `import type { Func } from "./subdir/types.ts";

const func: Func = t => t;
`;

console.log(ts.preProcessFile(text, true));
```

**Expected behavior:**

That the type only import would be referenced somehow.  As of TypeScript 3.8, `ambientExternalModules` was introduced, so I can only assume that is where it should go, but it isn't populated.  In addition, `ambientExternalModules` is of type `string[]` unlike the other properties which are `FileReference[]`.

**Actual behavior:**

```js
{
  referencedFiles: [],
  typeReferenceDirectives: [],
  libReferenceDirectives: [],
  importedFiles: [],
  isLibFile: false,
  ambientExternalModules: undefined
}
```

**Comment by RyanCavanaugh** [comment]
Also pinging @andrewbranch for opinions on what the best behavior here is

**Comment by sheetalkamat** [comment]
Its definitely bug and it should return importedFiles to contain the actual file form `import type`
